Code Analysis.md

```diff
@@ -110,12 +110,11 @@ According to the [CWE-916](https://cwe.mitre.org/data/definitions/916.html), the
 
-### B321	FTP-related functions are being called. FTP is considered insecure. Use SSH/SFTP/SCP or some other encrypted protocol.
-
 ### B401	A telnet-related module is being imported.  Telnet is considered insecure. Use SSH or some other encrypted protocol.
 Telnet is an outdated form of communication that does not use authentication to access a shell. [CWE-311](https://cwe.mitre.org/data/definitions/311.html) explains the weakness being; the software does not encrypt sensitive or critical information before storage or transmission. The lack of encryption could mean breaches of confidentiality, integrity, and accountability. The file [device_tracker.py](https://github.com/home-assistant/core/blob/dev/homeassistant/components/actiontec/device_tracker.py) uses telnet to get data for an Actiontec MI424WR (Verizon FIOS) router starting on line 93. After further research, the Actiontec MI424WR (Verizon FIOS) routers are outdated and do not support SSH. The only [work around](https://forums.verizon.com/t5/Fios-Internet/Remotely-restart-FIOS-MI424WR-Router/td-p/546593) found for safe remote access is to use an SSH device on the network to access the router. However, this solution is not feasible for Home Assistant, and the router is so outdated, there are likely little to none in use. For these reasons, so we decided not to submit a pull request. 
 
-### B402	A FTP-related module is being imported.  FTP is considered insecure. Use SSH/SFTP/SCP or some other encrypted protocol.
+### B321	A FTP-related module is being called. FTP is considered insecure. Use SSH/SFTP/SCP or some other encrypted protocol.
+FTP (File Transfer Protocol) is a standard networking protocol used for the transfer of files between a server and client across a network. [CWE-220](https://cwe.mitre.org/data/definitions/220) explains that the usage of an FTP server itself is a weakness, as the server stores sensitive data underneath the root directory that is accessible to the server's users. This grants files accessible to unauthorized actors in the system. The file [xiaomi/camera.py](https://github.com/home-assistant/core/blob/dev/homeassistant/components/xiaomi/camera.py) uses FTP to retrieve the latest video file from the Xiaomi FTP server starting on line 95. Upon [further reasearch](https://www.home-assistant.io/integrations/xiaomi/), this usage of FTP is intentional, as in order to maintain the ability to use the Xiaomi mobile app to view the live camera feed as well as the Home Assistant Integration. Usage of newer more secure technologies such as RTSP like most of the Home Assistant integrations would block the ability to use the Xiaomi mobile app features.
```

Comment:
Good point bringing up how RTSP would break the integration for this device. HA is tricky because these components are often not always designed by the developers of HA. Personally I would not use Xiaomi but it is important to note this issue so people can be made aware of it. Nice little write. Approved after you fix the header error.